source/geod24/Registry.d

```diff
@@ -0,0 +1,106 @@
+/*******************************************************************************
+
+    Registry implementation for multi-threaded access
+
+    This registry allows to look up a `Tid` based on a `string`.
+    It is extracted from the `std.concurrency` module to make it reusable
+
+*******************************************************************************/
+
+module geod24.Registry;
+
+import core.sync.mutex;
+import geod24.concurrency;
+
+/// Ditto
+public shared struct Registry
+{
+    private Tid[string] tidByName;
+    private string[][Tid] namesByTid;
+    private Mutex registryLock;
+
+    /// Initialize this registry, creating the Mutex
+    public void initialize() @safe nothrow
+    {
+        this.registryLock = new shared Mutex;
+    }
+
+    /**
+     * Gets the Tid associated with name.
+     *
+     * Params:
+     *  name = The name to locate within the registry.
+     *
+     * Returns:
+     *  The associated Tid or Tid.init if name is not registered.
+     */
+    Tid locate(string name)
+    {
+        synchronized (registryLock)
+        {
+            if (shared(Tid)* tid = name in this.tidByName)
+                return *cast(Tid*)tid;
+            return Tid.init;
+        }
+    }
+
+    /**
+     * Associates name with tid.
+     *
+     * Associates name with tid in a process-local map.  When the thread
+     * represented by tid terminates, any names associated with it will be
+     * automatically unregistered.
+     *
+     * Params:
+     *  name = The name to associate with tid.
+     *  tid  = The tid register by name.
+     *
+     * Returns:
+     *  true if the name is available and tid is not known to represent a
+     *  defunct thread.
+     */
+    bool register(string name, Tid tid)
+    {
+        synchronized (registryLock)
+        {
+            if (name in tidByName)
+                return false;
+            /*
+              TODO: This check was in `std.concurrency` but `isClosed` is private
```

Comment:
But it's in `geod24.*` now, you could make it public. ^^

Reply:
`package`, right